packages/headless/src/features/result-templates/result-templates-helpers.ts

```diff
@@ -10,7 +10,7 @@ import {Result} from '../../api/search/search/result';
  */
 export const getResultProperty = (result: Result, property: string) => {
   if (property in result) {
-    return (result as Record<string, unknown>)[property];
+    return ((result as unknown) as Record<string, unknown>)[property];
```

Comment:
This additional cast is needed after switching `Result` from a `type` to an `interface`.

Reply:
Wouldn't using `return result[property];` here work?

Reply:
Typescript raises an error because the `Result` interface does not have an index signature (`[key:string]: unknown`). Since property is of type `string` (not restricted to `keyof Result`), it causes an error. This is not the case with the raw object below which does have an index signature.

Reply:
Ah, my bad, I was confused with the `raw` field.